EventsExpress.Core/Services/AuthService.cs

```diff
@@ -1,128 +1,248 @@
         {
-            if (userDto == null)
+            var userDto = GetCurrentUser(userClaims);
+            var authLocal = userDto.Account.AuthLocal;
+            if (authLocal == null)
             {
-                throw new EventsExpressException("User not found");
+                throw new EventsExpressException("Invalid user");
             }
 
-            var jwtToken = _tokenService.GenerateAccessToken(userDto);
-            var refreshToken = _tokenService.GenerateRefreshToken();
+            if (!VerifyPassword(userDto.Account.AuthLocal, oldPassword))
+            {
+                throw new EventsExpressException("Invalid password");
+            }
 
-            // save refresh token
-            userDto.RefreshTokens = new List<RefreshToken> { refreshToken };
+            authLocal.Salt = PasswordHasher.GenerateSalt();
+            authLocal.PasswordHash = PasswordHasher.GenerateHash(newPassword, authLocal.Salt);
+            await Context.SaveChangesAsync();
+        }
 
-            await _userService.Update(userDto);
+        public async Task<bool> CanRegister(string email)
+        {
+            return !(await Context.AuthLocal.AnyAsync(al => al.Email == email));
+        }
 
-            return new AuthenticateResponseModel(jwtToken, refreshToken.Token);
+        public async Task<Guid> Register(RegisterDto registerDto)
+        {
+            var account = Mapper.Map<Account>(registerDto);
+            account.AccountRoles = new[] { new AccountRole { RoleId = Db.Enums.Role.User } };
+            var result = Insert(account);
+
+            await Context.SaveChangesAsync();
+
+            await _mediator.Publish(new RegisterVerificationMessage(account.AuthLocal));
+
+            return result.Id;
         }
 
-        public async Task ChangePasswordAsync(UserDto userDto, string oldPassword, string newPassword)
+        public async Task RegisterComplete(RegisterCompleteDto registerCompleteDto)
         {
-            if (VerifyPassword(userDto, oldPassword))
+            await _userService.Create(Mapper.Map<UserDto>(registerCompleteDto));
+        }
+
+        public UserDto GetCurrentUser(ClaimsPrincipal userClaims)
+        {
+            Claim emailClaim = userClaims.FindFirst(ClaimTypes.Name);
+
+            if (!Guid.TryParse(emailClaim.Value, out Guid userId))
             {
-                userDto.Salt = PasswordHasher.GenerateSalt();
-                userDto.PasswordHash = PasswordHasher.GenerateHash(newPassword, userDto.Salt);
-                await _userService.Update(userDto);
-                return;
+                throw new EventsExpressException("User not found");
             }
 
-            throw new EventsExpressException("Invalid password");
+            return _userService.GetById(userId);
         }
 
-        public UserDto GetCurrentUser(ClaimsPrincipal userClaims)
+        public async Task<UserDto> GetCurrentUserAsync(ClaimsPrincipal userClaims)
```

Comment:
Why do you need two same methods? Maybe remove one? For example leave async version only.